exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/interpreter/InterpreterEvaluator.java

```diff
@@ -378,8 +388,14 @@ public ValueHolder visitDoubleConstant(ValueExpressions.DoubleExpression dExpr,
     }
 
     @Override
-    public ValueHolder visitQuotedStringConstant(ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
-      return ValueHolderHelper.getVarCharHolder(getManagedBufferIfAvailable(), e.value);
+    public ValueHolder visitQuotedStringConstant(final ValueExpressions.QuotedString e, Integer value) throws RuntimeException {
+      return getConstantValueHolder(e.value, new Function<DrillBuf, ValueHolder>() {
```

Comment:
@arina-ielchiieva , I'm not sure if the change makes sense or not.

1. If I have a decimal value, decVal.   Let's say decVal.getBigDecimal().toString() return '123'.  When I call  ValueHolderHelper.getDecimal38Holder(buffer, decExpr.getBigDecimal().toString()), I should return a decimal holder.

2. Now, let's say I have a QuotedString '123', I call ValueHolderHelper.getVarCharHolder(buffer, e.value), since the key e.value is same as '123', and the constant value cache will just return the decimal holder, while I should get a VarcharHolder in stead.  Will it cause problem here?